exercises/palindrome-products/examples/success-standard/src/Palindromes.hs

```diff
@@ -25,9 +25,16 @@ union2 cmp merge = go
       EQ -> merge x y : go xs ys
       LT -> x : go xs (y:ys)
 
-largestPalindrome, smallestPalindrome :: Integral a => a -> a -> Product2 a
-largestPalindrome a b = head $ bigPalindromes a b
-smallestPalindrome a b = head $ smallPalindromes a b
+largestPalindrome, smallestPalindrome :: Integral a => a -> a -> Maybe (Product2 a)
+largestPalindrome a b | a > b = Nothing
```

Comment:
(small) I think the `a > b` is unnecessary, since both `bigPalindromes` and `smallPalindromes` will act on empty lists. Thus I'd rather see them removed,; no need to check things that we know are handled already. Ties in well with another change I have to suggest to these functions.

Examples if `a = 10` and `b = 5`:

```
Prelude> [10 .. 5]
[]
Prelude> [5, 4 .. 10]
[]
```